mllib/src/test/scala/org/apache/spark/ml/classification/DecisionTreeClassifierSuite.scala

```diff
@@ -232,6 +232,13 @@ class DecisionTreeClassifierSuite extends SparkFunSuite with MLlibTestSparkConte
     compareAPIs(rdd, dt, categoricalFeatures = Map.empty[Int, Int], numClasses)
   }
 
+  test("copied model must have the same parent") {
```

Comment:
I don't think you need to write a new unit test for each model.  A more efficient way would be to write a generic testing utility like this (maybe inside a new file like test/scala/org/apache/spark/ml/util/MLTestingUtils.scala):
```
object MLTestingUtils {
  def checkCopy(model: Model[_]): Unit = {
    val copied = model.copy()
    assert(copied.parent === model.parent)
  }
}
```
Later on, we can add other standard checks here as well.